clippy_lints/src/methods/iter_overeager_cloned.rs

```diff
@@ -47,13 +61,33 @@ pub(super) fn check<'tcx>(
             expr.span,
             msg,
             |diag| {
-                let method_span = expr.span.with_lo(cloned_call.span.hi());
-                if let Some(mut snip) = snippet_opt(cx, method_span) {
-                    snip.push_str(trailing_clone);
-                    let replace_span = expr.span.with_lo(cloned_recv.span.hi());
-                    diag.span_suggestion(replace_span, "try", snip, Applicability::MachineApplicable);
+                match op {
+                    Op::RmCloned | Op::LaterCloned => {
+                        let method_span = expr.span.with_lo(cloned_call.span.hi());
+                        if let Some(mut snip) = snippet_opt(cx, method_span) {
+                            snip.push_str(trailing_clone);
+                            let replace_span = expr.span.with_lo(cloned_recv.span.hi());
+                            diag.span_suggestion(replace_span, "try", snip, Applicability::MachineApplicable);
+                        }
+                    }
+                    Op::FixClosure(name, closure) => {
+                        if let Some(new_closure) = transform_closure(cx, closure) {
+                            // closure :      &T -> bool
+                            // new_closure : &&T -> bool
+                            let snip = format!(".{name}({new_closure}).cloned()" );
+                            let replace_span = expr.span.with_lo(cloned_recv.span.hi());
+                            diag.span_suggestion(replace_span, "try", snip, Applicability::MachineApplicable);
+                        }
+                    }
                 }
             }
         );
     }
 }
+
+fn transform_closure<'tcx>(cx: &LateContext<'tcx>, closure: Expr<'tcx>) -> Option<String> {
```

Comment:
Is there a case where this `None` cases could have some significance? Or it shouldn't never be `None`? In the case it can be `None` it should be accounted in the `match` in the lint emitter closure.